scheme.py

```diff
@@ -164,12 +165,26 @@ def to_slist(x: Sequence[SExp]) -> SList:
     return acc
 
 
+def make_bool(x: bool) -> SSym:
+    """
+    Returns a scheme boolean.
+
+    >>> make_bool(True)
+    SSym(name='true')
+    >>> make_bool(False)
+    SSym(name='false')
+    """
+    if x:
+        return SSym('true')
+    return SSym('false')
+
+
 @dataclass
 class SFunction(SExp):
     name: SSym
     formals: SList
     body: SList
-
+    code: Optional[bytecode.Function] = None
```

Comment:
This is supposed to be the primary compiled version of the function (either lazily or eagerly compiled).

We could change how that's stored later when we add specialization, or right now?

Reply:
To make sure I understand the flow you have in mind: The frontend compiles all the functions (assigning the bytecode to the `code` attribute), and then calls `run` on the `main` function?

Reply:
It compiles some number of functions, including `main`, and then the interpreter can call compile functions on-demand before a `CallInst` if they aren't pre-compiled. (It currently has an `UnimplementedException` there because we don't have the compiler.)